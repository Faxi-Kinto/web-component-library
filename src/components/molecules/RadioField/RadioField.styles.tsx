import styled from 'styled-components';
import { size } from 'styles/basics';
import { pxToRem } from 'styles/fonts';
import { defaultFontSize } from 'styles/constants';
import { flexRow } from 'styles/flex';
import { theme } from 'config';

export const Container = styled.label`
  position: relative;
  background: white;
  ${flexRow('initial', 'center')};
  font-size: ${pxToRem(defaultFontSize)};
  height: ${pxToRem('45px')};
  width: 100%;
  border: ${pxToRem('0.5px')} solid ${theme.pallet.DISABLED};
  box-sizing: border-box;
  cursor: pointer;

  input {
    display: none;
    &:checked {
      & + svg {
        .inner {
          display: block;
        }
        & + span {
          font-weight: 600;
        }
      }
    }
  }

  > svg {
    color: ${theme.pallet.TURQUOISE};
    margin: 0 ${pxToRem('16px')};

    &.lg {
      ${size(pxToRem('25px'))}
    }

    &.md {
      ${size(pxToRem('16px'))}
    }

    .inner {
      display: none;
      animation-name: animation-inner;
      animation-duration: 0.35s;
    }

    @keyframes animation-inner {
      0% {
        r: 45%;
      }
      100% {
        r: 30%;
      }
    }
  }
`;
