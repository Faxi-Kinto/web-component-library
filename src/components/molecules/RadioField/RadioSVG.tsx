import React from 'react';

type Props = {
  className?: string;
};

const RadioSVG: React.FC<Props> = (props: Props) => (
  <svg className={props.className} width="25" height="25" viewBox="0 0 25 25">
    <circle
      cx="12.5"
      cy="12.5"
      r="11.5"
      strokeWidth="1"
      stroke="currentColor"
      fill="none"
    />
    <circle className="inner" fill="currentColor" cx="12.5" cy="12.5" r="30%" />
  </svg>
);

export default RadioSVG;
