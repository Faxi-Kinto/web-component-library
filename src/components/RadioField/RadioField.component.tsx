import React from 'react';
import * as Styled from './RadioField.styles';
import Text from '../Text';
import RadioSVG from './RadioSVG';

export type SizeType = 'md' | 'lg';
export type RadioFieldProps = {
  value: string;
  name: string;
  label?: string;
  disabled?: boolean;
  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
  checked?: boolean;
  size?: SizeType;
  className?: string;
  inputRef?: React.RefObject<HTMLInputElement>;
};

const RadioField: React.FC<RadioFieldProps> = (
  props: RadioFieldProps
): JSX.Element => {
  const {
    label,
    name,
    disabled,
    checked,
    size,
    value,
    onChange,
    className,
    inputRef,
  } = props;

  return (
    <Styled.Container className={className}>
      <input
        ref={inputRef}
        className="radio-input"
        type="radio"
        name={name}
        disabled={disabled}
        checked={checked}
        value={value}
        onChange={(event: React.ChangeEvent<HTMLInputElement>): void => {
          onChange && onChange(event);
        }}
      />
      <RadioSVG className={size} />
      <Text.Body>{label}</Text.Body>
    </Styled.Container>
  );
};

RadioField.defaultProps = {
  disabled: false,
  size: 'lg',
};

export default RadioField;
