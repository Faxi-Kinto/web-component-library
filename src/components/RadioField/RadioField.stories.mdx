
import { Meta, Story, Preview, Props } from "@storybook/addon-docs/blocks";
import RadioField from "./"

<Meta title="Core|Molecules/RadioField" />

# RadioField Component

This is an implementation of the Radio button

## Implementation

#### Default
<Preview>
  <Story name="default">
    <RadioField name="default1" value="1" label="Default radio button" />
  </Story>
</Preview>

#### Large size
<Preview>
  <Story name="With large size">
    <RadioField name="default2" value="1" label="Large radio button" size="lg"/>
  </Story>
</Preview>

##### Medium size
<Preview>
  <Story name="With medium size">
    <RadioField name="default3" value="1" label="Medium radio button" size="md" />
  </Story>
</Preview>

## Props  
<Props of={RadioField} />
