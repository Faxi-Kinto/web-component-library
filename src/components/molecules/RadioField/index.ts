import RadioField from './RadioField.component';

export default RadioField;
