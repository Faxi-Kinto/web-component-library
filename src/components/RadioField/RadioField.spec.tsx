import React from 'react';
import renderer from 'react-test-renderer';

import RadioField from './RadioField.component';

test('RadioField renders correctly', () => {
  const component = renderer.create(<RadioField name="test" value="test" />);

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
