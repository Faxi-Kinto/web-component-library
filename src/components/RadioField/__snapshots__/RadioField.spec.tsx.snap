// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioField renders correctly 1`] = `
.c0 {
  position: relative;
  background: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: initial;
  -webkit-justify-content: initial;
  -ms-flex-pack: initial;
  justify-content: initial;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1rem;
  height: 2.8125rem;
  width: 100%;
  border: 0.03125rem solid #95A1A6;
  box-sizing: border-box;
  cursor: pointer;
}

.c0 input {
  display: none;
}

.c0 input:checked + svg .inner {
  display: block;
}

.c0 input:checked + svg + span {
  font-weight: 600;
}

.c0 > svg {
  color: #81c8d1;
  margin: 0 1rem;
}

.c0 > svg.lg {
  width: 1.5625rem;
  height: 1.5625rem;
}

.c0 > svg.md {
  width: 1rem;
  height: 1rem;
}

.c0 > svg .inner {
  display: none;
  -webkit-animation-name: animation-inner;
  animation-name: animation-inner;
  -webkit-animation-duration: 0.35s;
  animation-duration: 0.35s;
}

.c1 {
  padding: 0;
  margin: 0;
  font-size: 1rem;
  color: #4E606A;
  font-weight: 300;
  white-space: pre-wrap;
}

<label
  className="c0"
>
  <input
    className="radio-input"
    disabled={false}
    name="test"
    onChange={[Function]}
    type="radio"
    value="test"
  />
  <svg
    className="lg"
    height="25"
    viewBox="0 0 25 25"
    width="25"
  >
    <circle
      cx="12.5"
      cy="12.5"
      fill="none"
      r="11.5"
      stroke="currentColor"
      strokeWidth="1"
    />
    <circle
      className="inner"
      cx="12.5"
      cy="12.5"
      fill="currentColor"
      r="30%"
    />
  </svg>
  <span
    className="c1"
    color="#4E606A"
    fontSize="16px"
  />
</label>
`;
